### PR TITLE
Use entry points to support plugin mechanism.

### PR DESCRIPTION
**Pull request by shanchenqi**
As we have discussed in  Issue [#496](https://github.com/chaoss/grimoirelab/issues/496), we implemented [grimoirelab-elk-gitee](https://github.com/chaoss/grimoirelab-elk-gitee) as a plugin. In order to support this plugin solution, we need to use entry points as I created in this pull request.

### CODE REVIEW

File: releases/unreleased/Use_entry_points_to_support_plugin_mechanism.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+title: 'Use entry points to support plugin mechanism'
```
Comment (sduenas) [inline releases/unreleased/Use_entry_points_to_support_plugin_mechanism.yml:1]:
```suggestion
title: Entry points to support enriching plugin mechanism
```

File: releases/unreleased/Use_entry_points_to_support_plugin_mechanism.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+category: New feature
+author: ChenqiShan <chenqishan337@gmail.com>, LinHaiming <lhming23@outlook.com>
+issue: 
+notes: The plugin solution is used to incubate a new data source backend solution.
```
Comment (sduenas) [inline releases/unreleased/Use_entry_points_to_support_plugin_mechanism.yml:5]:
```suggestion
notes: >
    This plugin mechanism allows ELK to use third-party enriching backends.
    Developers only need to include the next snippet in their package configuration
    and ELK will import them as any other core backend. 
    ```
    entry_points={"grimoire_elk": "mybackend = package.subpackage:get_connectors"},
    ```
    **NOTE**: In the previous example, `get_connectors` is a function that returns
    the backend classes needed to enrich a datasource. Take the next snipped
    as an example:
    ```
    def get_connectors():
        return {"backend": [MyBackend, MyBackendOcean, MyBackendEnrich, MyBackendCommand]}
    ```
```

**Review by sduenas — CHANGES_REQUESTED** [review]
Thanks for your contribution! I've been testing the PR and works fine, however, will you mind to review my comments? Also, you should improve the commit message of this PR. See our [contributing guidelines](https://github.com/chaoss/grimoirelab/blob/master/CONTRIBUTING.md).

**Comment by shanchenqi** [comment]
Hi, @sduenas, Thank you for your code review. I have updated my pull request according to your advice.

**Comment by shanchenqi** [comment]
Excuse me, I force to push the commits again in order to squash the commits into one.

**Review by sduenas — CHANGES_REQUESTED** [review]
LGTM but, would you mind to change the message of the commit? Follow the guidelines I mentioned before (you can also use part of the text included in the changelog entry). Thanks!

**Comment by shanchenqi** [comment]
> LGTM but, would you mind to change the message of the commit? Follow the guidelines I mentioned before (you can also use part of the text included in the changelog entry). Thanks!

OK, I got it and updated the code. Thank you~

**Review by sduenas — APPROVED** [review]
LGTM. Thank you so much for your contribution. This feature will improve a lot the core of the platform.

**Comment by sduenas** [comment]
@shanchenqi last thing, I realized the PR doesn't contain the latest changes of the repo history. Would you mind to rebase it with master before I merge it?

**Comment by sduenas** [comment]
@shanchenqi thanks for the PR!

**Comment by shanchenqi** [comment]
> @shanchenqi thanks for the PR!

It's my pleasure!

**Pull request merged**